core/src/main/groovy/noe/rhel/server/workspace/WorkspaceHttpdTomcatsBaseOS.groovy

```diff
@@ -46,22 +47,51 @@ class WorkspaceHttpdTomcatsBaseOS extends WorkspaceMultipleTomcats{
     httpd.updateConfSetBindAddress(hostIpAddress)
 
     copyModulesIfMissing(httpd)
+    copyConfIfMissing(httpd)
     workspaceTomcat.prepare(true) //always true due to running with JWS 5 `ewsVersion >= new Version('3.1.0-DR0')`
 
     createAdditionalTomcatsWithRegistrations(numberOfAdditionalTomcats)
     originalTomcatHosts.putAll(originalTomcatHostsIpAddresses())
   }
 
+  /**
+   * JBCS modules are provided in different folder and need to be properly loaded or copied for smooth working.
+   * @param httpd Instance of `Httpd` server
+   */
   private void copyModulesIfMissing(Httpd httpd) {
-    File modulePath = new File("${httpd.getServerRoot()}/modules/mod_jk.so")
-    File sclModulePath = new File("${DefaultProperties.HTTPD_SCL_ROOT}/usr/lib64/httpd/modules/mod_jk.so")
-    if (!modulePath.exists() && sclModulePath.exists()) {
-      log.info("Copiing file ${sclModulePath} to modules folder of httpd ${modulePath}.")
-      modJkModules.add(modulePath)
-      JBFile.copyFile(sclModulePath, modulePath)
+    List<String> moduleNames = ["mod_jk", "mod_advertise", "mod_cluster_slotmem", "mod_manager", "mod_proxy_cluster"]
+    moduleNames.each() { String moduleName ->
+      File modulePath = new File("${httpd.getServerRoot()}/modules/${moduleName}.so")
+      File sclModulePath = new File("${DefaultProperties.HTTPD_SCL_ROOT}/usr/lib64/httpd/modules/${moduleName}.so")
+      if (!modulePath.exists() && sclModulePath.exists()) {
+        log.info("Copying file ${sclModulePath} to modules folder of httpd ${modulePath}.")
+        moduleFiles.add(modulePath)
+        JBFile.copyFile(sclModulePath, modulePath)
+      }
+      if (!modulePath.exists()) {
+        throw new FileNotFoundException("Module file is missing. ${modulePath}")
+      }
     }
-    if (!modulePath.exists()) {
-      throw new FileNotFoundException("Mod_jk module file is missing. ${modulePath}")
+  }
+
+  /**
+   * JBCS modules are provided in different folder then BaseOS httpd expects them and have to be copied for sake of
+   * mod_cluster tests
+   * @param httpd Instance of `Httpd` server
+   */
+  private void copyConfIfMissing(Httpd httpd) {
+    List<String> confRelativePaths = ["conf.d/mod_cluster.conf"]
+    confRelativePaths.each() { String confRelativePath->
+      File confPath = new File(httpd.getServerRoot(), confRelativePath)
+      File sclConfPath = new File("${DefaultProperties.HTTPD_SCL_ROOT}/etc/httpd/${confRelativePath}")
+      if (!confPath.exists() && sclConfPath.exists()) {
+        log.info("Copying file ${sclConfPath} to config.d folder of httpd ${confPath}.")
```

Comment:
logging...

Reply:
I don't see any logging in `FileStateVault.push()`, do I missing something?

Reply:
I think it is not relevant in this case because logging message is about copying not about saving state.

Please check comment above.